File: ruby/fizz_buzz/fizz_buzz.rb
class FizzBuzz
  def self.simple
    result = []

    (1..100).each do |number|
      if number % 15 == 0
        result << 'FizzBuzz'
      elsif number % 3 == 0
        result << 'Fizz'
      elsif number % 5 == 0
        result << 'Buzz'
      else
        result << number
      end
    end

    puts result.join ', '
  end
end

File: practice_java/src/main/java/practice/FizzBuzz.java
package practice;

public class FizzBuzz {
	public static String[] perform(int limit) {
		String[] result = new String[limit];

		for (int i = 1; i <= limit; i++) {
			String subResult;

			if (i % 15 == 0) {
				subResult = "FizzBuzz";
			} else if(i % 3 == 0) {
				subResult = "Fizz";
			} else if(i % 5 == 0) {
				subResult = "Buzz";
			} else {
				subResult = "" + i;
			}
			
			result[i - 1] = subResult;
		}
		
		return result;
	}
}
